### ISSUE TITLE
Caching never occurs when `~.copy()` function is called

### ISSUE THREAD

**Issue by TRoboto** [ManimCommunity/manim#1179]
## Description of bug / unexpected behavior

I noticed that the `copy.deepcopy` function prevents caching. I believe it does so because it produces objects with different `id` every time.

## Expected behavior

Caching should work as long as nothing changes in the scene. 

## How to reproduce the issue

<summary>Code for reproducing the problem</summary>

```py
class TestCopy(Scene):
    def construct(self):
        square = Square()
        square.generate_target().shift(3 * LEFT)
        self.play(MoveToTarget(square))
        self.wait()
```

**Comment by behackl** [comment]
CC @ManimCommunity/scene-caching

**Comment by huguesdevimeux** [comment]
Thanks for the report ! 

Scene caching is getting really old and needs maintenance. I saw that a an `id` has been added to every (alll ?) mobjects, which makes Scene-caching broken. I will try to investigate more on that.

**Comment by huguesdevimeux** [comment]
This has been addressed by #1211.
